### PR TITLE
Check service provider (EXPOSUREAPP-10865)

### PR DESCRIPTION
**Pull request by chiljamgossow**
after https://github.com/corona-warn-app/cwa-app-android/pull/4478

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt [hunk 1]
```diff
@@ -34,6 +47,12 @@ class DccTicketingQrCodeHandler @Inject constructor(
         )
     }
 
+    private fun Set<DccTicketingServiceProviderAllowListEntry>.validateServiceIdentity(serviceIdentity: String) {
+        if (!qrCodeSettings.checkServiceIdentity.value) return
+        val hash = serviceIdentity.toSHA256().encodeUtf8()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt:52]: `enodeUtf8()` isn't working as expected , please use `serviceIdentity.encode().sha256()`?
```suggestion
        val hash = serviceIdentity.encode().sha256()
```

Currently  the comparison is failing 
```diff
- serviceIdentityHash=[text=8646c015ffec92cfec73220f602c4892267c054f0eca4aff4d1b40238d394561]
+ ServiceProviderAllowList=[DccTicketingServiceProviderAllowListEntry(serviceIdentityHash=[hex=8646c015ffec92cfec73220f602c4892267c054f0eca4aff4d1b40238d394561])]
find=null
```
hashes are the same ,but the key is different `text=..` ,`hash=..`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt [hunk 2]
```diff
@@ -34,6 +47,12 @@ class DccTicketingQrCodeHandler @Inject constructor(
...
+        find { it.serviceIdentityHash == hash } ?: throw DccTicketingException(SP_ALLOWLIST_NO_MATCH)
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt:53]: Please some logs would be helpful ?

**Review by mtwalli — DISMISSED** [review]
LGTM, Tested with my suggestion and works fine

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt (hunk 2 above)
Comment (chiljamgossow) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/qrcode/DccTicketingQrCodeHandler.kt]: now it is chatty

**Review by mtwalli — APPROVED** [review]
LGTM

**Review by jurajkusnier — APPROVED** [review]
LGTM

**Pull request merged**